backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```diff
@@ -615,24 +617,28 @@ public Specimen convertToSpecimen(ConvertToSpecimenProps props) {
     var specimen = new Specimen();
     specimen.setId(props.getId());
     specimen.addIdentifier().setValue(props.getIdentifier());
+
+    var collection = specimen.getCollection();
+    var collectionCodeableConcept = collection.getBodySite();
+    var collectionCoding = collectionCodeableConcept.addCoding();
+    collectionCoding.setSystem(SNOMED_CODE_SYSTEM);
+    collectionCoding.setCode(
+        Optional.of(props)
+            .map(ConvertToSpecimenProps::getCollectionCode)
+            .orElse(DEFAULT_LOCATION_CODE));
+    collectionCodeableConcept.setText(
+        Optional.of(props)
+            .map(ConvertToSpecimenProps::getCollectionName)
+            .orElse(DEFAULT_LOCATION_NAME));
```

Comment:
this is not in the AC - my bad!
can we please only use the default location name if we know we're using the default location code? if a location code was provided but no location name, let's just not set the text for the bodySite
that way we're not sending an incorrect location name if for some reason we have a specimen with a location code but no location name

to break down the cases:
collectionCode and collectionName provided - use those
collectionCode provided but not collectionName - use the collectionCode and don't set the text for the name
collectionCode not provided but collectionName provided - use the default values for both
neither provided - use the default values